src/newlines.cpp

```diff
@@ -5928,6 +5920,23 @@ void do_blank_lines(void)
          }
       }
 
+      // Control blanks before an access spec
+      if (  (options::nl_before_access_spec() > 0)
+         && (options::nl_before_access_spec() != pc->nl_count)
+         && chunk_is_token(next, CT_ACCESS))
+      {
+         log_rule_B("nl_before_access_spec");
+
+         // Don't add blanks after an open brace
+         if (  prev == nullptr
+            || (  !chunk_is_token(prev, CT_BRACE_OPEN)
+               && !chunk_is_token(prev, CT_VBRACE_OPEN)))
```

Comment:
While this is equivalent to the original code, I am wondering whether VBRACEs are actually possible before/after an access specifier. I'm quite sure that this is impossible for C/C++, but don't know enough about some of the other supported languages.

Reply:
Just found `chunk_is_not_token`, which is probably better than `!chunk_is_token`. However, both also test for `pc != nullptr`, which is always true in this case and doesn't need to be tested again. The compiler may be able to optimize this test away, however, this is a bit unclear from the (huge) assembly code blob of this file. Thus the question: Shall the refactoring be reverted to not rely on the compiler or is the slightly increased readability of `chunk_is_not_token` worth it?

Reply:
YES! Refactoring would be fine with **chunk_is_not_token**